packages/api/src/auth/index.ts

```diff
@@ -65,13 +65,17 @@ export const getAuthenticationContext = async ({
 
   // No `auth-provider` header means that the user is logged out,
   // and none of this auth malarky is required.
-  if (!type || !authDecoder) {
+  if (!type) {
     return undefined
   }
 
   const { schema, token } = parseAuthorizationHeader(event)
 
-  const authDecoders = Array.isArray(authDecoder) ? authDecoder : [authDecoder]
+  const authDecoders = Array.isArray(authDecoder)
+    ? authDecoder
+    : authDecoder
+    ? [authDecoder]
+    : []
```

Comment:
I'm pretty bad at reading these; can we opt for something more verbose but easier to read?

Reply:
I switched to regular if statements. Hope that's easier to read